### PR TITLE
Fix a couple of HTML parsing issues

### PR DESCRIPTION
**Pull request by nox**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/15987)

### CODE REVIEW

File: components/script/dom/servoparser/html.rs [hunk 1]
```diff
@@ -264,6 +265,14 @@ impl TreeSink for Sink {
         }
     }
 
+    fn is_mathml_annotation_xml_integration_point(&self, handle: JS<Node>) -> bool {
```
Comment (Ms2ger) [inline components/script/dom/servoparser/html.rs]: Please add a spec link.

**Review by Ms2ger — APPROVED** [review]

**Comment by Ms2ger** [comment]
r+ for real :)

**Comment by jdm** [comment]
@bors-servo: retry
#15988

**Comment by nox** [comment]
```
  ▶ CRASH [expected OK] /_mozilla/mozilla/mozbrowser/reload.html
  │ 
  │ VMware, Inc.
  │ Gallium 0.4 on softpipe
  │ 3.3 (Core Profile) Mesa 12.0.1
  │ no entry found for key (thread RenderBackend, at src/libcore/option.rs:785)
  │ 2017-03-16 11:53:54.872 servo[66151:734609785] Metadata.framework [Error]: couldn't get the client port
  │ stack backtrace:
  │    0:        0x10889b73e - backtrace::backtrace::trace::hef11bbe02813de13
  │    1:        0x10889be8c - backtrace::capture::Backtrace::new::hb5a725a088a2a2fc
  │    2:        0x1082fd880 - servo::main::_$u7b$$u7b$closure$u7d$$u7d$::h827fe0401500c65f
  │    3:        0x109d883b9 - std::panicking::rust_panic_with_hook::hdbc3bba6a9dc0bb9
  │    4:        0x109d88214 - std::panicking::begin_panic::ha249c774b9bf96e1
  │    5:        0x109d88182 - std::panicking::begin_panic_fmt::ha197daaf02a649ab
  │    6:        0x109d880e7 - rust_begin_unwind
  │    7:        0x109db0570 - core::panicking::panic_fmt::h7414cb5ce71ea361
  │    8:        0x109db05fd - core::option::expect_failed::h9213c09c0fe70c78
  │    9:        0x108680d6a - webrender::frame_builder::FrameBuilder::build::h9dfc3dbbe562f073
  │   10:        0x1086a1adb - webrender::render_backend::RenderBackend::render::h7d725c99525b370d
  │   11:        0x108695d97 - webrender::render_backend::RenderBackend::run::h586004cebbd294f6
  │   12:        0x108630ab8 - std::panicking::try::do_call::hfd5d9d1d1c64810e
  │   13:        0x109d892ea - __rust_maybe_catch_panic
  │   14:        0x10864842a - _$LT$F$u20$as$u20$alloc..boxed..FnBox$LT$A$GT$$GT$::call_box::hd4f6c63a2c1d06fd
  │   15:        0x109d84f44 - std::sys::imp::thread::Thread::new::thread_start::h4008e1859fbd98b8
  │   16:     0x7fff8afbb059 - _pthread_body
  │   17:     0x7fff8afbafd6 - _pthread_start
  │ ERROR:servo: no entry found for key
  │ called `Result::unwrap()` on an `Err` value: IoError(Error { repr: Custom(Custom { kind: NotFound, error: StringError("Bogus destination port.") }) }) (thread main, at src/libcore/result.rs:860)
  │ called `Result::unwrap()` on an `Err` value: IoError(Error { repr: Custom(Custom { kind: NotFound, error: StringError("Bogus destination port.") }) }) (thread LayoutThread PipelineId { namespace_id: PipelineNamespaceId(0), index: PipelineIndex(1) }, at src/libcore/result.rs:860)
  │ stack backtrace:
  │    0:        0x10889b73e - backtrace::backtrace::trace::hef11bbe02813de13
  │    1:        0x10889be8c - backtrace::capture::Backtrace::new::hb5a725a088a2a2fc
  │    2:        0x1082fd880 - servo::main::_$u7b$$u7b$closure$u7d$$u7d$::h827fe0401500c65f
  │    3:        0x109d883b9 - std::panicking::rust_panic_with_hook::hdbc3bba6a9dc0bb9
  │    4:        0x109d88214 - std::panicking::begin_panic::ha249c774b9bf96e1
  │    5:        0x109d88182 - std::panicking::begin_panic_fmt::ha197daaf02a649ab
  │    6:        0x109d880e7 - rust_begin_unwind
  │    7:        0x109db0570 - core::panicking::panic_fmt::h7414cb5ce71ea361
  │    8:        0x109cbcc0d - core::result::unwrap_failed::h06ffc6135155777a
  │    9:        0x109cbefb4 - webrender_traits::api::RenderApi::set_root_pipeline::h4c8db59d9ccbe28d
  │   10:        0x1082ee454 - _$LT$compositing..compositor..IOCompositor$LT$Window$GT$$GT$::handle_browser_message::h447fc95b85997a56
  │   11:        0x1082dd914 - _$LT$servo..Browser$LT$Window$GT$$GT$::handle_events::h6ec0bf6a2504dd12
  │   12:        0x1082fc935 - servo::main::h79d4fbe2866a21f5
  │   13:        0x109d88045 - std::panicking::try::do_call::h24a2756282b9a31c
  │   14:        0x109d892ea - __rust_maybe_catch_panic
  │   15:        0x109d88786 - std::rt::lang_start::hd19f94db0c6a490e
  │ ERROR:servo: called `Result::unwrap()` on an `Err` value: IoError(Error { repr: Custom(Custom { kind: NotFound, error: StringError("Bogus destination port.") }) })
  │ stack backtrace:
  │    0:        0x10889b73e - backtrace::backtrace::trace::hef11bbe02813de13
  │    1:        0x10889be8c - backtrace::capture::Backtrace::new::hb5a725a088a2a2fc
  │    2:        0x1082fd880 - servo::main::_$u7b$$u7b$closure$u7d$$u7d$::h827fe0401500c65f
  │    3:        0x109d883b9 - std::panicking::rust_panic_with_hook::hdbc3bba6a9dc0bb9
  │    4:        0x109d88214 - std::panicking::begin_panic::ha249c774b9bf96e1
  │    5:        0x109d88182 - std::panicking::begin_panic_fmt::ha197daaf02a649ab
  │    6:        0x109d880e7 - rust_begin_unwind
  │    7:        0x109db0570 - core::panicking::panic_fmt::h7414cb5ce71ea361
  │    8:        0x109cbcc0d - core::result::unwrap_failed::h06ffc6135155777a
  │    9:        0x109cbf2d5 - webrender_traits::api::RenderApi::set_root_display_list::h1d0bad4f7fc8e67d
  │   10:        0x10891d167 - layout_thread::LayoutThread::perform_post_style_recalc_layout_passes::h6425087ef34a0c83
  │   11:        0x108918f3f - layout_thread::LayoutThread::handle_reflow::h7bf7badad5007450
  │   12:        0x108911dd6 - layout_thread::LayoutThread::handle_request_helper::he49efae9efe5fc0b
  │   13:        0x10890de85 - _$LT$layout_thread..LayoutThread$u20$as$u20$layout_traits..LayoutThreadFactory$GT$::create::_$u7b$$u7b$closure$u7d$$u7d$::h0e9c40cdb2a7f772
  │   14:        0x1088c38b7 - std::panicking::try::do_call::h40fbd8249f097d43
  │   15:        0x109d892ea - __rust_maybe_catch_panic
  │   16:        0x1088d9e08 - _$LT$F$u20$as$u20$alloc..boxed..FnBox$LT$A$GT$$GT$::call_box::hc7f39c3648ed2a32
  │   17:        0x109d84f44 - std::sys::imp::thread::Thread::new::thread_start::h4008e1859fbd98b8
  │   18:     0x7fff8afbb059 - _pthread_body
  │   19:     0x7fff8afbafd6 - _pthread_start
  │ ERROR:servo: called `Result::unwrap()` on an `Err` value: IoError(Error { repr: Custom(Custom { kind: NotFound, error: StringError("Bogus destination port.") }) })
  │ Pipeline failed in hard-fail mode.  Crashing!
  │ servo(66151,0x118553000) malloc: *** error for object 0x7f85fa2a5130: double free
  │ *** set a breakpoint in malloc_error_break to debug
  │ thread panicked while processing panic. aborting.
  └ thread panicked while processing panic. aborting.
```

Probably an intermittent.

@bors-servo try

**Comment by SimonSapin** [comment]
Homu seems stuck on this PR.

@bors-servo r-

**Comment by SimonSapin** [comment]
It seems like homu is forever waiting to hear back from buildbot about this build that finished long ago. I don’t know how to salvage this PR other than opening a new one :/

In the meantime, keeping it closed to let the queue make progress.

**Pull request merged**